docs/contributing/committing.rst

```diff
@@ -42,6 +45,8 @@ You can fix up any small mistakes in the commits,
 such as typos and formatting, as part of the rebase.
 ``git rebase --interactive`` is an excellent tool for this job.
 
+Ideally, use this as an opportunity to squash the changes to a few commits, so each commit is making a single meaningful change (and not breaking anything). If this is not possible because of the nature of the changes, go with the least ugly.
```

Comment:
I think it could be interesting to specify this further than "least ugly". If we want to follow the "each commit does not break anything" rule, then PRs where a few commits isn't doable should be squashed to a single commit?